src/openms/include/OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmTreeGuided.h

```diff
@@ -0,0 +1,175 @@
+#include <OpenMS/DATASTRUCTURES/BinaryTreeNode.h>
+#include <OpenMS/APPLICATIONS/MapAlignerBase.h>
+#include "MapAlignmentAlgorithmIdentification.h"
+
+namespace OpenMS
+{
+  /**
+    @brief A map alignment algorithm based on peptide identifications from MS2 spectra.
+
+    ID groups with the same sequence in different maps represent points of correspondence between the maps. They are used to evaluate the distances between the maps for hierarchical clustering and form the basis for the alignment.
+    Only the best PSM per spectrum is considered as the correct identification.
+
+    For each pair of maps, the distance is determined based on the intersection of the contained identifications and hierarchically clustered using average linkage.
+    Following the tree, the maps are aligned, resulting in a transformed feature map that contains both the original and the transformed retention times.
+    As long as there are at least two clusters, the alignment is done as follows:\n
+    Of every pair of clusters, the one with the larger 10/90 percentile retention time range is selected as reference for the align() method of @ref OpenMS::MapAlignmentAlgorithmIdentification.
+    align() maps the median retention time of each ID group in the second cluster to the reference retention time of this group.
+    Cubic spline smoothing is used to convert this mapping to a smooth function.
+    Retention times in the second cluster are transformed to the reference scale by applying this function.
+    Additionally, the original retention times are stored in the meta information of each ID group.
+    The reference is combined with the transformed cluster.
+
+    The resulting map is used to extract transformation descriptions for each input map.
+    For each map cubic spline smoothing is used to convert the mapping to a smooth function.
+    Retention times of each map are transformed applying corresponding function.
+
+    @htmlinclude OpenMS_MapAlignmentAlgorithmTreeGuided.parameters
+
+    @ingroup MapAlignment
+
+  */
+  class OPENMS_DLLAPI MapAlignmentAlgorithmTreeGuided :
+    public DefaultParamHandler,
+    public ProgressLogger
+  {
+public:
+    /// Default constructor
+    MapAlignmentAlgorithmTreeGuided();
+
+    /// Destructor
+    ~MapAlignmentAlgorithmTreeGuided() override;
+
+    /**
+     * @brief Extract RTs given for individual features of each map, calculate distances for each pair of maps and cluster hierarchical using average linkage.
+     *
+     * @param feature_maps Vector of input maps (FeatureMap) whose distance is to be calculated.
+     * @param tree Vector of BinaryTreeNodes that will be computed
+     * @param maps_ranges Vector to store all sorted RTs of extracted identifications for each map in @p feature_maps; needed to determine the 10/90 percentiles
+    */
+    static void buildTree_(std::vector<FeatureMap>& feature_maps, std::vector<BinaryTreeNode>& tree, std::vector<std::vector<double>>& maps_ranges);
```

Comment:
the underscore usually indicates that this is a protected or private member. Here it is in public namespace

Reply:
I guess this might be because of tests?

Reply:
No. During the refactoring, I just missed adapting it. I'll fix it.